### ISSUE TITLE
`arduino-fwuploader` panics

### ISSUE THREAD

**Issue by umbynos** [arduino/arduino-fwuploader#150]
### Describe the problem

```
./arduino-fwuploader
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x824766]

goroutine 1 [running]:
debug/elf.(*Section).ReadAt(0xc000200000?, {0xc000210000?, 0x270?, 0x24?}, 0x40?)
	<autogenerated>:1 +0x26
archive/zip.readDirectoryEnd({0xa23b20, 0xc000138e80}, 0x210)
	/opt/hostedtoolcache/go/1.19.1/x64/src/archive/zip/reader.go:526 +0xf5
archive/zip.(*Reader).init(0xc000179a40, {0xa23b20?, 0xc000138e80}, 0x210)
	/opt/hostedtoolcache/go/1.19.1/x64/src/archive/zip/reader.go:97 +0x5c
archive/zip.NewReader({0xa23b20, 0xc000138e80}, 0x210)
	/opt/hostedtoolcache/go/1.19.1/x64/src/archive/zip/reader.go:90 +0x5e
github.com/daaku/go%2ezipexe.zipExeReaderElf({0xa246c0?, 0xc0000ae0c0}, 0xd758be)
	/home/runner/go/pkg/mod/github.com/daaku/go.zipexe@v1.0.0/zipexe.go:128 +0x8b
github.com/daaku/go%2ezipexe.NewReader({0xa246c0, 0xc0000ae0c0}, 0x0?)
	/home/runner/go/pkg/mod/github.com/daaku/go.zipexe@v1.0.0/zipexe.go:48 +0x98
github.com/daaku/go%2ezipexe.OpenCloser({0xc0000d62d0?, 0xc000115720?})
	/home/runner/go/pkg/mod/github.com/daaku/go.zipexe@v1.0.0/zipexe.go:30 +0x57
github.com/cmaglie/go%2erice.init.0()
	/home/runner/go/pkg/mod/github.com/cmaglie/go.rice@v1.0.3/appended.go:42 +0x65
```

### To reproduce

This seems to only affect linux [2.2.1](https://github.com/arduino/arduino-fwuploader/releases/tag/2.2.1) released binary

### Expected behavior

The binary should not crash

### Arduino Firmware Uploader version

`arduino-fwuploader Version: 2.2.1 Commit: 75bcf76`

### Operating system

Linux

### Operating system version

Ubuntu 18.04 (same also on archlinux)

### Issue checklist

- [X] I searched for previous reports in [the issue tracker](https://github.com/arduino/arduino-fwuploader/issues?q=)
- [X] I verified the problem still occurs when using the latest version
- [X] My report contains all necessary details

**Comment by umbynos** [comment]
The problem seems to be caused by a cobo:
- misconfiguration of the go version used in the release pipeline: https://github.com/arduino/arduino-fwuploader/blob/fb99d12dc62a4fd1e454f8b334457ea120e8d879/.github/workflows/release-go-crosscompile-task.yml#L13 and https://github.com/arduino/arduino-fwuploader/blob/85d7e4611884239f21e26ecc514a6dd9d7910c18/.github/workflows/publish-go-tester-task.yml#L28 The `^` means to use a version greater than the one specified.
```
Run actions/setup-go@v3
Setup go version spec ^1.16.2
Found in cache @ /opt/hostedtoolcache/go/1.19.1/x64
Added go to the path
Successfully set up Go version ^1.16.2
go version go1.19.1 linux/amd64
```
This causes the other workflows (like the integration test) to use a version different from the version used for the build, and this is why the CI did not catch the problem before the release.
- incompatibility with go 1.19, see #145

**Comment by umbynos** [comment]
fixed by #151
